### PR TITLE
JW-1256 oper tests

### PR DESCRIPTION
**Pull request by mclapinski**

### CODE REVIEW

File: src/base/queue_task.h [hunk 1]
```diff
@@ -213,9 +213,8 @@ class WorkQueue {
     }
 
     ~WorkQueue() {
+        tbb::mutex::scoped_lock lock(mutex_);
```
Comment (kaweue) [inline src/base/queue_task.h:216]: @michal-clapinski I think I will need some explanation here, probably I am missing the point

Comment (mclapinski) [reply src/base/queue_task.h:216]: Yes, definitely. See this https://github.com/codilime/contrail-controller/pull/53/files#diff-b4e33c663c999f07a3d40b674c50548f. I'm actually reverting a change I made before. I thought this previous change would fix error in `queue_task_test.cc` but it didn't. I'm hoping adding `WaitForIdle` will fix it once and for all.

Comment (kaweue) [reply src/base/queue_task.h:216]: @michal-clapinski Ok now I see the point. However this lonely mutex seems to be a bit weird.

**Review by kaweue — APPROVED** [review]

File: src/base/queue_task.h (hunk 1 above)
Comment (mclapinski) [reply src/base/queue_task.h:216]: Also the comment `// Shutdown() needs to be called before deleting` and then not calling it is a bit weird but I got used to it.

**Review by katrybacka — APPROVED** [review]

**Pull request merged**